### ISSUE TITLE
Demo code for Custom Menu Items Does Not Create hover states correctly. 

### ISSUE THREAD

**Issue by lgnashold** [mantinedev/mantine#7074]
### Dependencies check up

- [X] I have verified that I use latest version of all @mantine/* packages

### What version of @mantine/* packages do you have in package.json?

7.13.4

### What package has an issue?

@mantine/core

### What framework do you use?

Vite

### In which browsers you can reproduce the issue?

All

### Describe the bug

When trying to use a custom component for a menu item, the hover state is shared between all menu items with that custom component. For example, if you have a custom component like so (taken straight from the docs):

```typescript
function CorrectItem(props: React.ComponentPropsWithoutRef<"button">) {
  return (
    <button type="button" {...props}>
      My custom Menu item
    </button>
  );
}
```

And you attempt to use it multiple times in your Menu:
```typescript
<Menu.Dropdown>
        <Menu.Item component={CorrectItem} />
        <Menu.Item component={CorrectItem} />
      </Menu.Dropdown>
```

When you hover over one element in the menu, all the items have the hover styling applied. 

![image](https://github.com/user-attachments/assets/2d7c8b96-1905-4b2b-9abf-eb36528b4102)
 
So far, have only reproduced this with button, not with, <a>. 

### If possible, include a link to a codesandbox with a minimal reproduction

https://codesandbox.io/p/sandbox/mantine-react-template-forked-95qk7x
### Possible fix

I would guess this has to do with how the props are passed along to the custom component - state is being reused where it shouldn't be. 

### Self-service

- [x] I would be willing to implement a fix for this issue

**Comment by lgnashold** [comment]
In the code I shared, looks like it's related to the className. Inspecting the className, it's 
```
mantine-focus-auto m_99ac2aa1 mantine-Menu-item m_87cf2631 mantine-UnstyledButton-root
```

And the problem is the second value is the same for both items in the mantine dropdown, which is causing them to share hover styles it seems like.

**Comment by rtivital** [comment]
Fixed in [7.13.5](https://github.com/mantinedev/mantine/releases/tag/7.13.5). The documentation has been updated for Vite and Next.js.
